full-moon/src/tokenizer.rs

```diff
@@ -768,13 +772,17 @@ peg::parser! {
             / identifier()
 
         pub(crate) rule tokens() -> Vec<(RawToken, usize)>
-            = shebang:(shebang:shebang() pos:position!() {(shebang,pos)})?
+            = bom:(bom:utf8_bom() pos:position!() {(bom,pos)})?
+              shebang:(shebang:shebang() pos:position!() {(shebang,pos)})?
               body:( token:token() pos:position!() {(token,pos)})*
               {
                   let mut body = body;
                   if let Some(shebang) = shebang {
                       body.insert(0, shebang)
                   }
+                  if let Some(bom) = bom {
+                    body.insert(0, bom)
```

Comment:
Can you run Cargo fmt over this? This doesn't look right

If Cargo fmt can't pick this up (since it's in a macro), can you keep this consistent with existing indents?

Reply:
Also, this should probably be before the shebang insertion, right?

Reply:
> This doesn't look right

I see

> Also, this should probably be before the shebang insertion, right?

It doesn't because it must be BEFORE shebang

Reply:
Oh, right, I forgot this was insert.